src/api/wordstrboxrenderer.cpp

```diff
@@ -44,31 +44,32 @@ char* TessBaseAPI::GetWordStrBoxText(int page_number) {
       continue;
     }
 
-  if (res_it->IsAtBeginningOf(RIL_TEXTLINE)) {
-    if (!first_line) {
+    if (res_it->IsAtBeginningOf(RIL_TEXTLINE)) {
+      if (!first_line) {
         wordstr_box_str.add_str_int("\n\t ", right + 1);
         wordstr_box_str.add_str_int(" ", image_height_ - bottom);
         wordstr_box_str.add_str_int(" ", right + 5);
         wordstr_box_str.add_str_int(" ", image_height_ - top);
         wordstr_box_str.add_str_int(" ", page_num);  // row for tab for EOL
         wordstr_box_str += "\n";
-    } else {
+      } else {
         first_line = false;
+      }
+      // Use bounding box for whole line for WordStr
+      res_it->BoundingBox(RIL_TEXTLINE, &left, &top, &right, &bottom);
+      wordstr_box_str.add_str_int("WordStr ", left);
+      wordstr_box_str.add_str_int(" ", image_height_ - bottom);
+      wordstr_box_str.add_str_int(" ", right);
+      wordstr_box_str.add_str_int(" ", image_height_ - top);
+      wordstr_box_str.add_str_int(" ", page_num);  // word
+      wordstr_box_str += " #";
     }
-    // Use bounding box for whole line for WordStr
-    res_it->BoundingBox(RIL_TEXTLINE, &left, &top, &right, &bottom);
-    wordstr_box_str.add_str_int("WordStr ", left);
-    wordstr_box_str.add_str_int(" ", image_height_ - bottom);
-    wordstr_box_str.add_str_int(" ", right);
-    wordstr_box_str.add_str_int(" ", image_height_ - top);
-    wordstr_box_str.add_str_int(" ", page_num);  // word
-    wordstr_box_str += " #";
-  }
-  do { wordstr_box_str +=
-                 std::unique_ptr<const char[]>(res_it->GetUTF8Text(RIL_WORD)).get();
-        wordstr_box_str += " ";
-        res_it->Next(RIL_WORD);
-     } while (!res_it->Empty(RIL_BLOCK) && !res_it->IsAtBeginningOf(RIL_WORD));
+    do {
+      wordstr_box_str +=
+          std::unique_ptr<const char[]>(res_it->GetUTF8Text(RIL_WORD)).get();
+      wordstr_box_str += " ";
+      res_it->Next(RIL_WORD);
+    } while (!res_it->Empty(RIL_BLOCK) && !res_it->IsAtBeginningOf(RIL_WORD));
   }
   wordstr_box_str.add_str_int("\n\t ", right + 1);
```

Comment:
@Shreeshrii, `right`, `top` and `bottom` can be uninitialized here (reported by Coverity Scan). What would be the right fix for this?

Reply:
There is a similar problem in [lstmboxrenderer.cpp:82](https://github.com/tesseract-ocr/tesseract/blob/master/src/api/lstmboxrenderer.cpp#L82) and 3 more lines.

Reply:
@stweil For some reason I did not get any email regarding the comments on the commits. Got the msg re issue today.

The EOL tab needs to be added at end of a line. Currently it is identified via IsAtBeginningOf(RIL_TEXTLINE). However that would put an EOL tab box at the beginning of the page,  hence I am checking that it is NOT the first line on page.

There is probably a different way to structure the code to get rid of this problem.  Please change the code as you see fit. Thanks!